packages/wallet/src/util.ts

```diff
@@ -1 +1,143 @@
+import cbor from 'cbor';
+
+type Props = {
+  keyAgent: KeyAgent;
+  logger?: Logger;
+};
+
+export const createCip30WalletApiFromWallet = (wallet: SingleAddressWallet, props: Props): WalletApi => {
+  const logger = props.logger || dummyLogger;
+  return {
+    getBalance: async (): Promise<Cbor> => {
+      logger.debug('getting balance');
+      try {
+        const value = await firstValueFrom(wallet.balance.available$);
+        return Buffer.from(coreToCsl.value(value).to_bytes()).toString('hex');
+      } catch (error) {
+        logger.error(error);
+        throw error;
+      }
+    },
+    getChangeAddress: async (): Promise<Cbor> => {
+      logger.debug('getting changeAddress');
+      try {
+        const [{ address }] = await firstValueFrom(wallet.addresses$);
+        const parsedAddress = parseCslAddress(address.toString());
+
+        if (!address || !parsedAddress) {
+          logger.error('could not get change address');
+          throw new ApiError(500, 'could not get change address');
+        } else {
+          return Buffer.from(parsedAddress.to_bytes()).toString('hex');
+        }
+      } catch (error) {
+        logger.error(error);
+        throw new ApiError(500, error);
+      }
+    },
+    getNetworkId: async (): Promise<number> => {
+      logger.debug('getting networkId');
+      return Promise.resolve(props.keyAgent.networkId);
+    },
+    getRewardAddresses: async (): Promise<Cbor[]> => {
+      logger.debug('getting reward addresses');
+      try {
+        const [{ rewardAccount }] = await firstValueFrom(wallet.addresses$);
+        const parsedAddress = parseCslAddress(rewardAccount.toString());
+
+        if (!rewardAccount || !parsedAddress) {
+          throw new ApiError(500, 'could not get reward address');
+        } else {
+          return [Buffer.from(parsedAddress.to_bytes()).toString('hex')];
+        }
+      } catch (error) {
+        logger.error(error);
+        throw new ApiError(500, error);
+      }
+    },
+    getUnusedAddresses: async (): Promise<Cbor[]> => {
+      logger.debug('getting unused addresses');
+      return Promise.resolve([]);
+    },
+    getUsedAddresses: async (_paginate?: Paginate): Promise<Cbor[]> => {
+      logger.debug('getting changeAddress');
+
+      const [{ address }] = await firstValueFrom(wallet.addresses$);
+      const parsedAddress = parseCslAddress(address.toString());
+
+      if (!address || !parsedAddress) {
+        throw new ApiError(500, 'could not get used addresses');
+      } else {
+        return [Buffer.from(parsedAddress.to_bytes()).toString('hex')];
+      }
+    },
+    getUtxos: async (amount?: Cbor, paginate?: Paginate): Promise<Cardano.Utxo[] | undefined> => {
+      let utxos = await firstValueFrom(wallet.utxo.available$);
+
+      if (amount) {
+        try {
+          const filterAmount = CSL.Value.from_bytes(Buffer.from(amount, 'hex'));
+          const { inputSelection } = await wallet.initializeTx({
```

Comment:
Please add a comment to explain why we're initializing a transaction here, since it's not really self-explanatory.

"Getting UTxOs to meet a required amount is a complex operation, which is handled by input selection capabilities. By initializing a transaction we're able to utilise the internal configuration and algorithm to make this selection, using a wallet address to satisfy the interface only."